pkg/fanal/analyzer/language/rust/cargo/cargo_test.go

```diff
@@ -382,6 +382,133 @@ func Test_cargoAnalyzer_Analyze(t *testing.T) {
 			dir:  "testdata/sad",
 			want: &analyzer.AnalysisResult{},
 		},
+		{
+			name: "workspace members",
+			dir:  "testdata/toml-workspace-members",
+			want: &analyzer.AnalysisResult{
+				Applications: []types.Application{
+					{
+						Type:     types.Cargo,
+						FilePath: "Cargo.lock",
+						Libraries: types.Packages{
+							{
+								ID:       "aho-corasick@1.1.2",
+								Name:     "aho-corasick",
+								Version:  "1.1.2",
+								Indirect: true,
+								Locations: []types.Location{
+									{
+										StartLine: 5,
+										EndLine:   12,
+									},
+								},
+								DependsOn: []string{"memchr@2.6.4"},
+							},
+							{
+								ID:       "gdb-command@0.7.6",
+								Name:     "gdb-command",
+								Version:  "0.7.6",
+								Indirect: false,
```

Comment:
I don't have a lot of exp with Cargo, can you help me understand:
i have played with `cargo tree` command.
And it looks like `cargo` builds separate tree for each members:
```bash
root@e5d322bb6f9e:/app2/casr# cargo tree

casr v2.9.0 (/app2/casr/casr)
├── anyhow v1.0.75
├── chrono v0.4.31
│   ├── iana-time-zone v0.1.58
...
libcasr v2.9.0 (/app2/casr/libcasr) (*)
```


Perhaps we need to add members as libs or mark all dependencies from members as Indirect?

Reply:
According to [cargo docs](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html#creating-a-workspace):

```
A workspace is a set of packages that share the same Cargo.lock and output directory. 
```

So, workspace is not a package, it couldn't be a dependency. For example CASR workspace has to crates (packages):
casr: [crates.io/crates/casr](https://crates.io/crates/casr)
libcasr: [crates.io/crates/libcasr](https://crates.io/crates/libcasr)
They are considered as separate crates. The same picture I see for LibAFL [github.com/AFLplusplus/LibAFL](https://github.com/AFLplusplus/LibAFL).
I think, the way we process members is right, they are Direct dependencies

Reply:
Okay, thanks for the explanation!